snowplow_tracker/emitter_configuration.py

```diff
@@ -0,0 +1,17 @@
+from typing import Optional, Union, Tuple
+from snowplow_tracker.typing import SuccessCallback, FailureCallback
+
+class EmitterConfiguration(object):
```

Comment:
Could we make use of Python properties in order to define the public API of the configuration class? I think that would make it easier to inspect what options the class has to offer. You can [learn more about properties here](https://realpython.com/python-property/).

Also don't forget to add comments everywhere since this public API. For most of these properties and classes you can copy nice comments from the Java tracker or elsewhere, it's rarely necessary to write new API docs.

As an example how we could use the properties:

```python
class EmitterConfiguration:
    def __init__(
            self,
            buffer_size: Optional[int] = None,
            on_success: Optional[SuccessCallback] = None,
            on_failure: Optional[FailureCallback] = None,
            byte_limit: Optional[int] = None,
            request_timeout: Optional[Union[float, Tuple[float, float]]] = None) -> None:
        """
            Configuration for the emitter that sends events over network to the Snowplow collector.
            :param  buffer_size:          Number of events to buffer in one request. Defaults to XX.
            :type   buffer_size:          int
            ....
        """
        self.buffer_size = buffer_size
        self.on_success = on_success
        self.on_failure = on_failure
        self.byte_limit = byte_limit
        self.request_timeout = request_timeout

    @property
    def buffer_size(self):
        """
        Number of events to buffer in one request. Defaults to XX.
        """
        return self._buffer_size

    @buffer_size.setter
    def buffer_size(self, value):
         self._buffer_size = float(value)
```